test/e2e_env/kubernetes/gateway/gatewayapi.go

```diff
@@ -31,79 +25,71 @@ spec:
-mesh: default
+mesh: %s
 metadata:
   name: external-service
 spec:
   tags:
     kuma.io/service: external-service
     kuma.io/protocol: http
   networking:
-    address: external-service.external-services.svc.cluster.local:80
-`
-
-	ExternalServicesNamespace := "external-services"
-	err := NewClusterSetup().
-		Install(GatewayAPICRDs).
-		Install(Kuma(config_core.Standalone,
-			WithCtlOpts(map[string]string{"--experimental-gatewayapi": "true"}),
-		)).
-		Install(NamespaceWithSidecarInjection(TestNamespace)).
-		Install(testserver.Install(
-			testserver.WithName("test-server-1"),
-			testserver.WithNamespace(TestNamespace),
-			testserver.WithEchoArgs("echo", "--instance", "test-server-1"),
-		)).
-		Install(testserver.Install(
-			testserver.WithName("test-server-2"),
-			testserver.WithNamespace(TestNamespace),
-			testserver.WithEchoArgs("echo", "--instance", "test-server-2"),
-		)).
-		Install(Namespace(ExternalServicesNamespace)).
-		Install(testserver.Install(
-			testserver.WithName("external-service"),
-			testserver.WithNamespace(ExternalServicesNamespace),
-			testserver.WithEchoArgs("echo", "--instance", "external-service"),
-		)).
-		Install(YamlK8s(externalService)).
-		Install(YamlK8s(GatewayClass)).
-		Setup(cluster)
-	Expect(err).ToNot(HaveOccurred())
-
-	E2EDeferCleanup(func() {
-		Expect(cluster.DeleteNamespace(TestNamespace)).To(Succeed())
-		Expect(cluster.DeleteNamespace(ExternalServicesNamespace)).To(Succeed())
-		Expect(cluster.DeleteKuma()).To(Succeed())
-		Expect(cluster.DismissCluster()).To(Succeed())
+    address: external-service.gatewayapi-external-services.svc.cluster.local:80
+`, meshName)
+
+	BeforeAll(func() {
+		setup := NewClusterSetup().
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(Namespace(externalServicesNamespace)).
+			Install(MTLSMeshKubernetes(meshName)).
+			Install(testserver.Install(
+				testserver.WithName("test-server-1"),
+				testserver.WithMesh(meshName),
+				testserver.WithNamespace(namespace),
+				testserver.WithEchoArgs("echo", "--instance", "test-server-1"),
+			)).
+			Install(testserver.Install(
+				testserver.WithName("test-server-2"),
+				testserver.WithMesh(meshName),
+				testserver.WithNamespace(namespace),
+				testserver.WithEchoArgs("echo", "--instance", "test-server-2"),
+			)).
+			Install(testserver.Install(
+				testserver.WithName("external-service"),
+				testserver.WithNamespace(externalServicesNamespace),
+				testserver.WithEchoArgs("echo", "--instance", "external-service"),
+			)).
+			Install(YamlK8s(externalService)).
+			Install(YamlK8s(gatewayClass))
+		Expect(setup.Setup(env.Cluster)).To(Succeed())
 	})
-})
 
-func GatewayAPI() {
-	if Config.IPV6 {
-		return // KIND which is used for IPV6 tests does not support load balancer that is used in this test.
-	}
+	E2EAfterAll(func() {
+		Expect(env.Cluster.DeleteNamespace(namespace)).To(Succeed())
+		Expect(env.Cluster.DeleteNamespace(externalServicesNamespace)).To(Succeed())
```

Comment:
Change this to `TriggerDeleteNamespace`. We don't want to wait for the namespace to be deleted.

Reply:
Indeed, missed that